src/main/java/seedu/mindmymoney/Parser.java

```diff
@@ -1,46 +1,40 @@
 package seedu.mindmymoney;
 
+import seedu.mindmymoney.command.AddCommand;
 import seedu.mindmymoney.command.Command;
 import seedu.mindmymoney.command.HelpCommand;
+import seedu.mindmymoney.constants.Indexes;
+import seedu.mindmymoney.helper.Functions;
 
 /**
  * Represents the input parser and deals with making sense of user commands.
  */
 public class Parser {
     public static final int SPLIT_LIMIT = 2;
-    public static final int COMMAND_INDEX = 0;
 
     protected String inputCommand;
 
     public Parser(String inputCommand) {
         this.inputCommand = inputCommand;
     }
 
-    /**
-     * Separates the user input into the command and description for easy reference.
-     *
-     * @return String array of user input.
-     */
-    public String[] parseInput() {
-        String[] inputAsArray = inputCommand.split(" ", SPLIT_LIMIT);
-        return inputAsArray;
-    }
-
     /**
      * Returns a Command object with respect to their input. The command object can then be executed to perform
      * the said command.
      *
      * @return Command object with respect to user's input.
      */
     public Command parseCommand() {
-        String[] parsedInput = parseInput();
-
-        switch (parsedInput[COMMAND_INDEX]) {
+        String[] parsedInput = Functions.parseInput(inputCommand);
+        switch (parsedInput[Indexes.INDEX_OF_FIRST_ITEM_IN_STRING]) {
         case "help":
             return new HelpCommand(true);
         case "bye":
             System.out.println("Goodbye!");
             System.exit(0);
+            return new HelpCommand(false); //solving fall through issue, need return something leh
```

Comment:
I think we can just leave it here first. I was hoping when you guys add on more cases the issue disappears...

Reply:
ok.... was trying to clear the build checks... iritating thing